media/src/iosMain/kotlin/DefaultSoundPlayer.kt

```diff
@@ -0,0 +1,221 @@
+/*
+ Copyright 2024 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.media
+
+import com.splendo.kaluga.media.DefaultSoundPlayer.State.Foreground
+import kotlinx.cinterop.CPointer
+import kotlinx.cinterop.ObjCObjectVar
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.pointed
+import kotlinx.cinterop.ptr
+import kotlinx.cinterop.value
+import platform.AVFAudio.AVAudioEngine
+import platform.AVFAudio.AVAudioPlayerNode
+import platform.AVFAudio.AVAudioFile
+import platform.AVFAudio.AVAudioSession
+import platform.AVFAudio.AVAudioSessionCategoryPlayback
+import platform.AVFAudio.AVAudioSessionRouteChangeNotification
+import platform.Foundation.NSBundle
+import platform.Foundation.NSError
+import platform.Foundation.NSNotification
+import platform.Foundation.NSNotificationCenter
+import platform.Foundation.NSNotificationName
+import platform.Foundation.NSOperationQueue.Companion.currentQueue
+import platform.Foundation.NSOperationQueue.Companion.mainQueue
+import platform.Foundation.NSURL
+import platform.UIKit.UIApplicationDidEnterBackgroundNotification
+import platform.UIKit.UIApplicationWillEnterForegroundNotification
+
+/**
+ * A default implementation of [SoundPlayer]
+ * @param source the [MediaSource.Local] on which the media is found
+ * @param configuration the [Configuration] of [SoundPlayer]
+ */
+actual class DefaultSoundPlayer(source: MediaSource.Local, private val configuration: Configuration) : SoundPlayer {
+    /**
+     * The [Configuration] of [SoundPlayer]
+     */
+    data class Configuration(
+        /**
+         * The player will be restarted if the route changed.
+         * Practically it means that the player will be updated if user connect/ disconnects audio outputs so the media can be played.
+         * Note apple doesn't recommend to play on speaker if user unplugged headphones (https://developer.apple.com/documentation/avfaudio/responding-to-audio-route-changes)
+         */
+        val restartIfRouteChanged: Boolean = true,
```

Comment:
should it not be `false` by default then?

Reply:
We are using it with the option "true". But I think you are right, it should be false by default

Reply:
updated